pkg/redact/redact.go

```diff
@@ -0,0 +1,62 @@
+// Copyright © 2020 by PACE Telematics GmbH. All rights reserved.
+// Created at 2020/12/16 by Florian S.
+
+package redact
+
+import (
+	"regexp"
+)
+
+type PatternRedactor struct {
+	patterns []*regexp.Regexp
+	scheme   RedactionScheme
+}
+
+// NewPatternRedactor creates a new redactor for masking certain patterns
+func NewPatternRedactor(scheme RedactionScheme) *PatternRedactor {
+	return &PatternRedactor{
+		scheme: scheme,
+	}
+}
+
+func (r *PatternRedactor) Mask(data string) string {
+	for _, pattern := range r.patterns {
+		if pattern == nil {
```

Comment:
Why do we even allow for pattern to be nil, is there a purpose for this?

Reply:
Technically one could add a nil pattern using the `AddPatterns`.

Reply:
Resolving this issue, because thinking about it again, it's a technicality and should not pose a real word problem